core/src/forest/ForestTrainers.cpp

```diff
@@ -54,11 +54,12 @@ ForestTrainer instrumental_trainer(double reduced_form_weight,
 }
 
 ForestTrainer multi_causal_trainer(size_t num_treatments,
-                                   size_t num_outcomes) {
+                                   size_t num_outcomes,
+                                   size_t response_length) {
```

Comment:
Instead of adding a new parameter, could this method just compute `response_length = num_treatments * num_outcomes` internally?

Reply:
Yes, but I left it slightly redundant like this for now, as this allows us (Stefan!) to more easily experiment with more funky stuff later, like splitting on an intercept, gives response_length = num_treatments * num_outcomes + 1, stuff like that :P

Reply:
Going ahead with the merge - assuming you are fine with this minor redundancy for the sake of future experimentation ease!